testdata/grammar_outside/elmamun.gap

```diff
@@ -0,0 +1,203 @@
+}
+
+algebra alg_enum auto enum;
+algebra alg_count auto count;
+
+algebra alg_pretty implements sig_elmamun(alphabet=char, answer=Rope) {
+  Rope number(int value) {
+    Rope res;
+    append(res, value);
+    return res;
+  }
+  Rope add(Rope left, char opSymbol, Rope right) {
+    Rope res;
+    append(res, '(');
+    append(res, left);
+    append(res, opSymbol);
+    append(res, right);
+    append(res, ')');
+    return res;
+  }
+  Rope heinz(Rope left, Rope opSymbol, Rope right) {
+    Rope res;
+    append(res, '(');
+    append(res, left);
+    append(res, opSymbol);
+    append(res, right);
+    append(res, ')');
+    return res;
+  }
+  Rope mult(Rope left, char opSymbol, Rope right) {
+    Rope res;
+    append(res, '(');
+    append(res, left);
+    append(res, opSymbol);
+    append(res, right);
+    append(res, ')');
+    return res;
+  }
+  Rope minus(Rope left, char opSymbol, Rope right) {
+    Rope res;
+    append(res, '(');
+    append(res, left);
+    append(res, opSymbol);
+    append(res, right);
+    append(res, ')');
+    return res;
+  }
+  Rope nil(void) {
+    Rope res;
+    return res;
+  }
+  choice [Rope] h([Rope] candidates) {
+    return candidates;
+  }
+}
+
+algebra alg_buyer implements sig_elmamun(alphabet=char, answer=int) {
+  int number(int value) {
+    return value;
+  }
+  int add(int left, char opSymbol, int right) {
+    return left + right;
+  }
+  int heinz(int left, Rope opSymbol, int right) {
+    return left + right;
+  }
+  int mult(int left, char opSymbol, int right) {
+    return left * right;
+  }
+  int minus(int left, char opSymbol, int right) {
+    return left - right;
+  }
+  int nil(void) {
+    return 0;
+  }
+  choice [int] h([int] candidates) {
+    return list(minimum(candidates));
+  }
+}
+algebra alg_seller extends alg_buyer {
+  choice [int] h([int] candidates) {
+    return list(maximum(candidates));
+  }
+}
+
+algebra alg_time implements sig_elmamun(alphabet=char, answer=int) {
+  int number(int value) {
+    return 0;
+  }
+  int add(int left, char opSymbol, int right) {
+    if (left > right) {
+      return left + 2;
+    } else {
+      return right + 2;
+    }
+  }
+  int heinz(int left, Rope opSymbol, int right) {
+    if (left > right) {
+      return left + 2;
+    } else {
+      return right + 2;
+    }
+  }
+  int mult(int left, char opSymbol, int right) {
+    if (left > right) {
+      return left + 5;
+    } else {
+      return right + 5;
+    }
+  }
+  int minus(int left, char opSymbol, int right) {
+    if (left > right) {
+      return left + 3;
+    } else {
+      return right + 3;
+    }
+  }
+  int nil(void) {
+    return 0;
+  }
+  choice [int] h([int] candidates) {
+    return list(minimum(candidates));
+  }
+}
+
+grammar gra_elmamun uses sig_elmamun(axiom = formula) {
+  formula = number(INT)
+	  | add(formula, CHAR, formula)
+	  | mult(formula, CHAR('*'), formula)
+	  | heinz(formula, ROPE("manfred"), formula)
+	  | nil(EMPTY)
```

Comment:
you have the algebra function "minus" but did not define it here

Reply:
yepp. Legal but unintentionally. But this file was really just used to test `ROPE("manfred")`